src/dev/impl/DevToys/ViewModels/Tools/Generators/CheckSumGenerator/CheckSumGeneratorToolViewModel.cs

```diff
@@ -0,0 +1,429 @@
+
+        private void ExecuteCancelCommand()
+        {
+            lock (_lockObject)
+            {
+                _hasCancelledCalculation = true;
+                _cancellationTokenSource.Cancel();
+                _cancellationTokenSource.Dispose();
+
+                _cancellationTokenSource = new CancellationTokenSource();
+
+                if (_outputHashingAlgorithm != null)
+                {
+                    InputHashingAlgorithm = _outputHashingAlgorithm;
+                }
+
+                _isCalculationInProgress = false;
+                _hasCancelledCalculation = false;
+                ShouldDisplayProgress = false;
+                ResetProgress();
+            }
+        }
+
+        #endregion
+
+        private void UpdateOutputCase()
+        {
+            if (string.IsNullOrWhiteSpace(Output))
+            {
+                return;
+            }
+
+            if (IsUppercase)
+            {
+                Output = Output!.ToUpperInvariant();
+            }
+            else
+            {
+                Output = Output!.ToLowerInvariant();
+            }
+        }
+
+        private void CheckSum()
+        {
+            lock (_lockObject)
+            {
+                ResetProgress();
+                CheckSumAsync().Forget();
+            }
+        }
+
+        private async Task CheckSumAsync()
+        {
+            if (_isCalculationInProgress || InputFile is null)
+            {
+                return;
+            }
+
+            _isCalculationInProgress = true;
+
+            string? hashOutput = await CalculateFileHash(InputHashingAlgorithm.Value, InputFile);
+
+            ThreadHelper.RunOnUIThreadAsync(() =>
+            {
+                if (hashOutput != null)
+                {
+                    Output = hashOutput;
+                    _outputHashingAlgorithm = InputHashingAlgorithm;
+                }
+
+                _isCalculationInProgress = false;
+                ShouldDisplayProgress = false;
+
+                if (!_toolSuccessfullyWorked)
+                {
+                    _toolSuccessfullyWorked = true;
+                    _marketingService.NotifyToolSuccessfullyWorked();
+                }
+            }).ForgetSafely();
+        }
+
+        private async Task<string?> CalculateFileHash(HashingAlgorithm inputHashingAlgorithm, IStorageFile? inputFile)
+        {
+            try
+            {
+                using HashAlgorithm? hashAlgo = CreateHashAlgorithm(inputHashingAlgorithm);
+                Stream? fileStream = await inputFile.OpenStreamForReadAsync();
+
+                ShouldDisplayProgress = HashingHelper.ComputeHashIterations(fileStream) > _displayAboveIterations;
+
+                byte[]? fileHash = await HashingHelper.ComputeHashAsync(
+                        hashAlgo,
+                        fileStream,
+                        new Progress<HashingProgress>(UpdateProgress),
+                        _cancellationTokenSource.Token);
```

Comment:
I see that you use a Stream but I don't see the dispose of this stream?

Reply:
Well spotted, missed that one 😁